pkg/receive/handler.go

```diff
@@ -74,48 +65,69 @@ func instrumentHandler(handlerName string, handler http.HandlerFunc) http.Handle
 	)
 }
 
+func NewHandlerInMux(mux *http.ServeMux, logger log.Logger, o *Options) *Handler {
```

Comment:
Consider removing this, 2 calls are more clear than having different constructor methods.